### DISCUSSION TITLE
New linter rule: explicit function return type

### DISCUSSION THREAD

**Discussion by Infir3** [biomejs/biome#540] [Ideas]
Hello,

I would like a rule that enforces an explicit function return type, like the one that typescript-eslint provides: https://typescript-eslint.io/rules/explicit-function-return-type/

Reasoning is the same as for the ESLint rule:

> Functions in TypeScript often don't need to be given an explicit return type annotation. Leaving off the return type is less code to read or write and allows the compiler to infer it from the contents of the function.
> 
> However, explicit return types do make it visually more clear what type is returned by a function. They can also speed up TypeScript type checking performance in large codebases with many large functions.

**Comment by ematipico** [comment]
I strongly suggest to chime in in this discussion: https://github.com/biomejs/biome/discussions/3

That's where we are discussing gathering rules from other sources :)

**Reply by Infir3** [reply]
Thanks, will do that :)
